backend/src/main/java/com/f12/moitz/domain/subway/SubwayRouteCalculator.java

```diff
@@ -0,0 +1,235 @@
+package com.f12.moitz.domain.subway;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.PriorityQueue;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class SubwayRouteCalculator {
+
+    private final SubwayEdges edges;
+
+    public SubwayRouteCalculator(final SubwayEdges edges) {
+        this.edges = edges;
+    }
+
+    public StationSequence findShortestTimePath(final SubwayStation start, final SubwayStation end) {
+        validateStationsExist(start, end);
+        validateNotSameStation(start, end);
+
+        final Map<SubwayStation, Integer> times = new HashMap<>();
+        final Map<SubwayStation, List<PreviousInfo>> prev = new HashMap<>();
+        final PriorityQueue<Node> pq = new PriorityQueue<>(Comparator.comparingInt(n -> n.time));
+        final Set<SubwayStation> visited = new HashSet<>();
+
+        times.put(start, 0);
+        pq.add(new Node(start, 0));
+
+        while (!pq.isEmpty()) {
+            final Node current = pq.poll();
+            // 현재 역 정보 가져오기
+            final SubwayStation currentStation = current.station;
+
+            if (!edges.containsStation(currentStation)) {
+                continue;
+            }
+            if (!visited.add(currentStation)) {
+                continue;
+            }
+            if (end.equals(currentStation)) {
+                break;
+            }
+
+            Set<Edge> currentEdges = edges.getEdges(currentStation);
+
+            for (Edge edge : currentEdges) {
+                final SubwayStation neighbor = edge.getDestination();
+
+                if (visited.contains(neighbor)) {
+                    continue;
+                }
+
+                final SubwayLine currentLine = edge.getSubwayLine();
+                int newTime = times.getOrDefault(currentStation, Integer.MAX_VALUE) + edge.getTimeInSeconds();
```

Comment:
반복적으로 나오는 매직 넘버를 상수화하는 건 어떤가요? 👀 

```suggestion
                private static final int UNREACHABLE_TIME = Integer.MAX_VALUE;

                int newTime = times.getOrDefault(currentStation, UNREACHABLE_TIME) + edge.getTimeInSeconds();

```